src/Cli/dotnet/commands/dotnet-test/MSBuildConnectionHandler.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.IO.Pipes;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.Test;
+
+namespace Microsoft.DotNet.Cli.commands.dotnet_test
+{
+    internal class MSBuildConnectionHandler : IDisposable
+    {
+        private readonly PipeNameDescription _pipeNameDescription = NamedPipeServer.GetPipeName(Guid.NewGuid().ToString("N"));
+        private readonly List<NamedPipeServer> _namedPipeConnections = new();
+        private readonly string[] _args;
+
+        private TestApplicationActionQueue _actionQueue;
+
+        public MSBuildConnectionHandler(string[] args, TestApplicationActionQueue actionQueue)
+        {
+            _args = args;
+            _actionQueue = actionQueue;
+        }
+
+        public async Task WaitConnectionAsync(CancellationToken token)
+        {
+            VSTestTrace.SafeWriteTrace(() => $"Wait for connection(s) on pipe = {_pipeNameDescription.Name}");
+
+            try
+            {
+                while (true)
```

Comment:
Avoids some uneeded throw/catch

```suggestion
                while (!token.IsCancellationRequested )
```